compiler/rustc_trait_selection/src/traits/select/mod.rs

```diff
@@ -1276,10 +1276,9 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
 
         // FIXME(generic_associated_types): Compare the whole projections
         let data_poly_trait_ref = projection_ty.map_bound(|proj| proj.trait_ref(self.tcx()));
-        let obligation_poly_trait_ref = obligation_trait_ref.to_poly_trait_ref();
```

Comment:
do we ever actually call this method with an `obligation_trait_ref` which contained unbound vars?

Cause if not I don't see why we would want to change this.

Reply:
Well, maybe, maybe not. The change here (and in other places), is that we were doing `skip_binder` only to call `to_poly_trait_ref` on that skipped value. I guess if there aren't bound vars here, then changing this to an explicit `Binder::dummy` is also valid, but that seems less clear overall; it feels better to me to *maintain* "potential" binders until we explicitly remove them via something like `replace_bound_vars_with_placeholders` or something.

Reply:
I mostly feel that using `Binder<T>` when `T` never actually has unbound vars is actual detrimental.

I currently prefer using `Binder::dummy` here. 

> but that seems less clear overall

At least for myself this change makes it less obvious what's going on :sweat_smile:

Reply:
>I mostly feel that using Binder<T> when T never actually has unbound vars is actual detrimental.
Well, in this case, are we sure that having bound vars here will never make sense?

I'm changed this to use `Binder::dummy` locally and am running tests. If everything passes, I'll switch back.